src/test/java/rx/observers/TestSubscriberTest.java

```diff
@@ -772,4 +772,47 @@ public void awaitValueCountFails() throws Exception {
         Assert.assertFalse(ts.awaitValueCount(5, 1, TimeUnit.SECONDS));
         
     }
+    
+    @Test
+    public void assertAndConsume() {
+        TestSubscriber<Integer> ts = TestSubscriber.create();
+        
+        ts.assertNoValues();
+        
+        ts.onNext(1);
+        
+        ts.assertAndConsume(1);
+        
+        ts.assertNoValues();
+        
+        ts.onNext(2);
+        ts.onNext(3);
+        
+        ts.assertValueCount(2);
+        
+        ts.assertAndConsume(2, 3);
+        
+        ts.onNext(4);
+        ts.onNext(5);
+        
+        try {
+            ts.assertAndConsume(4);
+            Assert.fail("Should have thrown AssertionError");
+        } catch (AssertionError ex) {
+            // expected
+        }
+        
+        ts.assertValueCount(2);
+        
+        try {
+            ts.assertAndConsume(4, 5, 6);
+            Assert.fail("Should have thrown AssertionError");
```

Comment:
In IntelliJ I usually just write `fail(` and hit `ALT + ENTER` to apply static import, IntellIJ applies it for whole file then